transport/tchannel/header.go

```diff
@@ -118,17 +102,36 @@ var emptyMap = map[string]string{}
 // InboundCallResponse.
 //
 // If the format is JSON, the headers are JSON encoded.
-func writeHeaders(format tchannel.Format, headers transport.Headers, getWriter func() (tchannel.ArgWriter, error)) error {
+func writeHeaders(format tchannel.Format, headers map[string]string, tracingBaggage map[string]string, getWriter func() (tchannel.ArgWriter, error)) error {
```

Comment:
Using separate maps for headers and tracing baggage gives us an opportunity to avoid allocating a new map with merge, by threading these through encodeHeaders or encodeJSONHeaders, but I’m willing to punt this optimization to help you regain momentum.